tests/t566-operator.c

```diff
@@ -0,0 +1,184 @@
+/// @file
+/// Test assembly of non-symmetric mass matrix operator (multi-component) see t537
+/// \test Test assembly of non-symmetric mass matrix operator (multi-component)
+#include <ceed.h>
+#include <stdlib.h>
+#include <math.h>
+#include "t566-operator.h"
+
+int main(int argc, char **argv) {
+  Ceed ceed;
+  CeedElemRestriction elem_restr_x, elem_restr_u,
+                      elem_restr_qd_i;
+  CeedBasis basis_x, basis_u;
+  CeedQFunction qf_setup, qf_mass;
+  CeedOperator op_setup, op_mass;
+  CeedVector q_data, X, U, V;
+  CeedInt P = 3, Q = 3, dim = 2, num_comp = 2;
+  CeedInt n_x = 1, n_y = 1;
+  CeedInt num_elem = n_x * n_y;
+  CeedInt num_dofs = (n_x*2+1)*(n_y*2+1), num_qpts = num_elem*Q*Q;
+  CeedInt ind_x[num_elem*P*P];
+  CeedScalar assembled[num_comp*num_comp*num_dofs*num_dofs];
+  CeedScalar x[dim*num_dofs], assembled_true[num_comp*num_comp*num_dofs*num_dofs];
+  CeedScalar *u;
+  const CeedScalar *v;
+
+  CeedInit(argv[1], &ceed);
+
+  // DoF Coordinates
+  for (CeedInt i=0; i<n_x*2+1; i++)
+    for (CeedInt j=0; j<n_y*2+1; j++) {
+      x[i+j*(n_x*2+1)+0*num_dofs] = (CeedScalar) i / (2*n_x);
+      x[i+j*(n_x*2+1)+1*num_dofs] = (CeedScalar) j / (2*n_y);
+    }
+  CeedVectorCreate(ceed, dim*num_dofs, &X);
+  CeedVectorSetArray(X, CEED_MEM_HOST, CEED_USE_POINTER, x);
+
+  // Qdata Vector
+  CeedVectorCreate(ceed, num_qpts, &q_data);
+
+  // Element Setup
+  for (CeedInt i=0; i<num_elem; i++) {
+    CeedInt col, row, offset;
+    col = i % n_x;
+    row = i / n_x;
+    offset = col*(P-1) + row*(n_x*2+1)*(P-1);
+    for (CeedInt j=0; j<P; j++)
+      for (CeedInt k=0; k<P; k++)
+        ind_x[P*(P*i+k)+j] = offset + k*(n_x*2+1) + j;
+  }
+
+  // Restrictions
+  CeedElemRestrictionCreate(ceed, num_elem, P*P, dim, num_dofs, dim*num_dofs,
+                            CEED_MEM_HOST, CEED_USE_POINTER, ind_x, &elem_restr_x);
+  CeedElemRestrictionCreate(ceed, num_elem, P*P, num_comp, num_dofs,
+                            num_comp*num_dofs,
+                            CEED_MEM_HOST, CEED_USE_POINTER, ind_x, &elem_restr_u);
+  CeedInt strides_qd[3] = {1, Q*Q, Q*Q};
+  CeedElemRestrictionCreateStrided(ceed, num_elem, Q*Q, 1, num_qpts, strides_qd,
+                                   &elem_restr_qd_i);
+
+  // Bases
+  CeedBasisCreateTensorH1Lagrange(ceed, dim, dim, P, Q, CEED_GAUSS, &basis_x);
+  CeedBasisCreateTensorH1Lagrange(ceed, dim, num_comp, P, Q, CEED_GAUSS,
```

Comment:
It'll be easier to debug when changing these to `CEED_GAUSS_LOBATTO`, which produces many fewer points in my test.